elastic/datadog_checks/elastic/elastic.py

```diff
@@ -78,9 +80,9 @@ def check(self, _):
             # retrieve the cluster name from the data, and append it to the
             # master tag list.
             cluster_name_tag = "cluster_name:{}".format(stats_data['cluster_name'])
-            self.config.tags.append(cluster_name_tag)
-            self.config.health_tags.append(cluster_name_tag)
-        self._process_stats_data(stats_data, stats_metrics)
+            base_tags.append(cluster_name_tag)
+            service_check_tags.append(cluster_name_tag)
```

Comment:
:+1: This was the origin of the leak, right?

Reply:
yup, there used to be `self.config.tags.append(` on each check run